packages/firestore/test/unit/specs/spec_builder.ts

```diff
@@ -457,12 +461,16 @@ export class SpecBuilder {
   writeAcks(
     doc: string,
     version: TestSnapshotVersion,
-    options?: { expectUserCallback: boolean }
+    options?: { expectUserCallback?: boolean; keepInQueue?: boolean }
   ): this {
     this.nextStep();
-    this.currentStep = { writeAck: { version } };
+    options = options || {};
+
+    this.currentStep = {
+      writeAck: { version, keepInQueue: !!options.keepInQueue }
```

Comment:
FWIW it may be worth adding a TODO(multitab) to update the spec schemas on the other clients with the changes you've made...  Just so we don't forget to do that before / soon-after merging multitab into master.